File: example-languages/src/main/java/com/abstratt/simon/examples/Tagged.java
package com.abstratt.simon.examples;

import java.util.Collection;

import com.abstratt.simon.metamodel.dsl.Meta;

/**
 * Small example metamodel exercising multivalued primitive slot values.
 */
@Meta.Package(builtIns = {})
public interface Tagged {

    @Meta.Composite(root = true)
    interface Item {
        @Meta.Name
        @Meta.Attribute
        String name();

        @Meta.Typed(String.class)
        @Meta.Attribute
        Collection<String> tags();
    }
}

File: example-languages-kotlin/src/main/kotlin/com/abstratt/simon/examples/kotlin/Tagged.kt
package com.abstratt.simon.examples.kotlin

import com.abstratt.simon.metamodel.dsl.Meta

@Meta.Package(builtIns = [])
interface Tagged {
    @Meta.Composite(root = true)
    interface Item {
        @Meta.Name
        @Meta.Attribute
        fun name(): String?

        @Meta.Typed(String::class)
        @Meta.Attribute
        fun tags(): Collection<String>?
    }
}
